config/application.rb

```diff
@@ -42,6 +43,18 @@ class Application < Rails::Application
       ENV[
         "DATABASE_URL"
       ] = "postgres://#{username}:#{password}@#{host}:#{port}/#{dbname}"
+    elsif ENV["DB_CREDENTIALS"].present?
+      # for environment which uses RDS aurora managed credentials only the the username
+      # and password is automaticallyset. The environment variable is then DB_CREDENTIALS
+      db_config = JSON.parse(ENV["DB_CREDENTIALS"])
+      username = CGI.escape(db_config["username"])
+      password = CGI.escape(db_config["password"])
+      host = CGI.escape(ENV["DB_HOST"])
+      dbname = CGI.escape(db_config["DB_NAME"])
+      port = 5432
```

Comment:
Would this ever come from an environment variable? Perhaps something like `ENV.fetch("DB_PORT", 5432)` might be good?